cmp/tile/TilingContainerModel.js

```diff
@@ -0,0 +1,219 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2021 Extremely Heavy Industries Inc.
+ */
+import {HoistModel} from '@xh/hoist/core';
+import {observable, computed, action, makeObservable} from '@xh/hoist/mobx';
+import {isEmpty} from 'lodash';
+
+/**
+ * Model for a TilingContainer.
+ *
+ * This model takes a collection of data objects, and provides a tiling layout to display
+ * them. Each `data` entry is passed to the `content` component in each tile - it is up to
+ * the content to determine how to render this `data`.
+ *
+ * It will generate a tiling layout that attempts to compromise between having as little empty
+ * space as possible, and keeping the tile width / height ratio as close to the desired ratio as possible.
+ *
+ * Supports a number of optional constraints on tile dimensions that can be used to produce more
+ * stable layouts. These should be used judiciously, however, as each constraint limits the ability
+ * of the TilingContainer to fill its available space.
+ */
+export class TilingContainerModel extends HoistModel {
+
+    @observable.ref
+    data;
+
+    @observable.ref
+    dimensions;
+
+    //-----------------------
+    // Immutable Properties
+    //-----------------------
+    content;
+    emptyText;
+    desiredRatio;
+    spacing;
+    minRatio;
+    maxRatio;
+    minWidth;
+    maxWidth;
+    minHeight;
+    maxHeight;
+
+    @computed.struct
```

Comment:
might put `layout` and `onResize` in special block above implementation, indicating they are for calling by the component only, not applications.  Might change name of `onResize` to something like `noteDimensions()` (rename less important)